sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.StartStop.cs

```diff
@@ -654,7 +657,8 @@ public async Task StopProcessingStopsProcessing(bool async)
                     }
 
                     completionSource.TrySetResult(true);
-                });
+                })
+                .Returns(Task.CompletedTask);
```

Comment:
I believe this may have been the cause of the flakiness.   Because the callback was invoked after the call returned, the processor would have already gotten unblocked on moving forward and may have set `stopInvoked` before the callback was executed.